trie.go

```diff
@@ -301,12 +301,52 @@ func makeMatcher(matcherType string, matcherArgs []string) (patternMatcher, erro
 	switch matcherType {
 	case "string":
 		return newStringMatcher(matcherArgs)
+	case "path":
+		return newPathMatcher(matcherArgs)
 	case "int":
 		return newIntMatcher(matcherArgs)
 	}
 	return nil, fmt.Errorf("unsupported matcher: %s", matcherType)
 }
 
+func newPathMatcher(args []string) (patternMatcher, error) {
+	if len(args) != 1 {
+		return nil, fmt.Errorf("expected only one parameter - variable name, got: %s", args)
+	}
+	return &pathMatcher{name: args[0]}, nil
+}
+
+type pathMatcher struct {
+	name string
+}
+
+func (s *pathMatcher) String() string {
+	return fmt.Sprintf("<path:%s>", s.name)
+}
+
+func (s *pathMatcher) getName() string {
+	return s.name
+}
+
+func (s *pathMatcher) match(i *charIter) bool {
+	s.grabValue(i)
+	return true
+}
+
+func (s *pathMatcher) equals(other patternMatcher) bool {
+	_, ok := other.(*stringMatcher)
```

Comment:
```suggestion
	_, ok := other.(*pathMatcher)
```

Maybe you want to handle `stringMatcher` and `pathMatcher`, but I'm not sure.